spec/controllers/results_controller_spec.rb

```diff
@@ -322,6 +322,34 @@ def self.test_unauthorized(route_name)
         expect(incomplete_result.overall_comment).to eq SAMPLE_COMMENT
       end
     end
+
+    context 'accessing an assignment with deductive annotations' do
+      it 'returns annotation data with criteria information' do
+        assignment = create(:assignment_with_deductive_annotations)
+        post :get_annotations, params: { assignment_id: assignment.id,
+                                         submission_id: assignment.groupings.first.current_result.submission.id,
+                                         id: assignment.groupings.first.current_result }, xhr: true
```

Comment:
Add `format: :json` to the params, and then down below you don't need to call `JSON.parse`, instead you can access `response.parsed_body`, which will be formatted for you.

Search the tests for `parsed_body` to see some examples of how we do this in some tests already.